axelor-base/src/main/java/com/axelor/apps/base/service/AdvancedExportServiceImpl.java

```diff
@@ -755,41 +747,50 @@ public MetaFile advancedExportCSV(MetaFile exportFile, MetaModel metaModel, List
-			AdvancedExportLine advancedExportLine = advancedExportLineRepo.find(Long.parseLong(fieldLine.get("id").toString()));
-			totalCols[i++] = advancedExportLine.getTargetField();
-		}
-		csvWriter.writeNext(totalCols);
-		
-		i = 0;
-		for (Map<String, Object> field : allFieldDataList) {
-			String[] allCols = field.keySet().toArray(new String[field.size()]);
-			Integer[] allColIndices = new Integer[allCols.length];
-			
-			for (int j = 0; j < allCols.length; j++) {
-				String col = allCols[j];
-				allColIndices[j] = Integer.parseInt(col.replace("Col_", ""));
-			}
-			Arrays.sort(allColIndices);
-			
-			for(Integer colIndex: allColIndices) {
-				String colName = "Col_" + String.valueOf(colIndex);
-				Object value = field.get(colName);
-				if (value == null || value == "") {
-					totalCols[i++] = null;
-				} else {
-					totalCols[i++] = value.toString();
-				}
-			}
-			csvWriter.writeNext(totalCols);
-			i = 0;
-		}
-		csvWriter.close();
-		
+        try (CSVWriter csvWriter = new CSVWriter(new FileWriter(csvFile, true), ';')) {
+            String[] totalCols = new String[advancedExportLines.size()];
+            int i = 0;
+
+            for (Map<String, Object> fieldLine : advancedExportLines) {
+                AdvancedExportLine advancedExportLine = advancedExportLineRepo
+                        .find(Long.parseLong(fieldLine.get("id").toString()));
+
+                String cellValue = "";
+                MetaTranslation metaTranslation = metaTranslationRepo.all().filter("self.key = ?1 and self.language = ?2",advancedExportLine.getTitle(), language).fetchOne();
+                if (metaTranslation != null && !metaTranslation.getMessage().equals("")) {
+                        cellValue = metaTranslation.getMessage();
+                } else {
+                    cellValue = advancedExportLine.getTitle();
+                }
+                totalCols[i++] = cellValue;
+            }
+            csvWriter.writeNext(totalCols);
+
+            i = 0;
+            for (Map<String, Object> field : allFieldDataList) {
+                String[] allCols = field.keySet().toArray(new String[field.size()]);
+                Integer[] allColIndices = new Integer[allCols.length];
+
+                for (int j = 0; j < allCols.length; j++) {
+                    String col = allCols[j];
+                    allColIndices[j] = Integer.parseInt(col.replace("Col_", ""));
+                }
+                Arrays.sort(allColIndices);
+
+                for (Integer colIndex : allColIndices) {
+                    String colName = "Col_" + colIndex;
+                    Object value = field.get(colName);
+                    if (value == null || value == "") {
+                        totalCols[i++] = null;
+                    } else {
+                        totalCols[i++] = value.toString();
+                    }
+                }
+                csvWriter.writeNext(totalCols);
+                i = 0;
+            }
```

Comment:
You have to close the csvWriter or call CsvTool class.

Reply:
A try-with-ressource is used, with it, we don't need to close csvWriter as it close it automatically.
Try-with-ressource was introduced in Java 7.
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html